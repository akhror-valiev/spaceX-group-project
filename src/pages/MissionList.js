import React from 'react';
import { useDispatch } from 'react-redux';
import { PropTypes } from 'prop-types';
import { Button, Badge } from 'react-bootstrap';
import { joinMission, leaveMission } from '../redux/missions/missions';
import './Missions.css';

const MissionList = ({
  name, description, reserved, id,
}) => {
  const dispatch = useDispatch();

  const handleJoinMission = (id) => {
    dispatch(joinMission(id));
  };

  const handleLeaveMission = (id) => {
    dispatch(leaveMission(id));
  };

  return (
    <tr className="mission-list">
      <td className="mission-name">{name}</td>
      <td className="mission-desc">{description}</td>
      <td className="mission">
        {
          reserved
            ? (<Badge className="badge-size info">Active Member</Badge>)
            : (<Badge className="badge-size secondary">Not A Member</Badge>)
        }
      </td>
      <td className="mission">
        {
          reserved
            ? (<Button onClick={() => handleLeaveMission(id)} className="leave-mission">Leave Mission</Button>)
            : (<Button onClick={() => handleJoinMission(id)} className="join-mission">Join Mission</Button>)
        }
      </td>
    </tr>
  );
};

export default MissionList;

MissionList.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
  id: PropTypes.string.isRequired,
};
