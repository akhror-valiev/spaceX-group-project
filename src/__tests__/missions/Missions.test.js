import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import MissionList from '../../pages/MissionList';

describe('snapshot testing', () => {
  test('snapshot for Mission component', () => {
    const rendererComponent = renderer
      .create(
        <Provider store={store}>
          <MissionList />
        </Provider>,
      )
      .toJSON();
    expect(rendererComponent).toMatchSnapshot();
  });
});
