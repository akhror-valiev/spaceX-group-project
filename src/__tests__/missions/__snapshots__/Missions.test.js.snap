// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot for Mission component 1`] = `
<tr
  className="mission-list"
>
  <td
    className="mission-name"
  />
  <td
    className="mission-desc"
  />
  <td
    className="mission"
  >
    <span
      className="badge-size secondary badge bg-primary"
    >
      Not A Member
    </span>
  </td>
  <td
    className="mission"
  >
    <button
      className="join-mission btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Join Mission
    </button>
  </td>
</tr>
`;
